code/src/GridAPI/Pagination.ts

```diff
@@ -65,15 +65,26 @@ namespace GridAPI.Pagination {
      *
      * @export
      * @param {string} gridID Id of the Grid from which to obtain the pagination Index
-     * @return {*}  {number} Index of the current page, 0 based. Return -1 if the Grid is not yet initialized.
+     * @return {*}  {string} Stringified JSON structure containing Index of the current page, error message and code, and success boolean
      */
-    export function GetCurrentPage(gridID: string): number {
+    export function GetCurrentPage(gridID: string): string {
         PerformanceAPI.SetMark('Pagination.GetCurrentPage');
+        let output = '';
 
-        if (!OSFramework.Helper.IsGridReady(gridID)) return -1;
-        const grid = GridManager.GetGridById(gridID);
+        if (!OSFramework.Helper.IsGridReady(gridID)) {
+            let returnMessage = new OSFramework.OSStructure.ReturnMessage();
+
+            returnMessage = {
+                value: -1,
+                isSuccess: false,
+                message: 'Grid not found',
+                code: OSFramework.Enum.ErrorCodes.CFG_GridNotFound
+            };
```

Comment:
I don't understand this code. First you create and instance, then you override that instance with a new object?...

Reply:
That's actually not overriding or is it? If you don't instantiate the ReturnMessage before you assign the values you won't have the intellisense

Reply:
`let returnMessage = new OSFramework.OSStructure.ReturnMessage() ` has no constructor, so it will actually create an empty object, then you just assign the new values by { key: value} taking advantage of the intellisense